first commit

LikeARivers opened the pull request.

LikeARivers committed:
first commit

AlexKitIt commented on conftest.py:
```diff
@@ -0,0 +1,56 @@
+    elif request.param == 'chrome':
+        browser = webdriver.Chrome()
+    browser.maximize_window()
+    #browser.get(urls.URL.STELLAR_BURGERS)
```
Нужно исправить: закомменченного кода быть не должно

AlexKitIt commented on conftest.py:
```diff
@@ -0,0 +1,56 @@
+@allure.step("Авторизация")
+@pytest.fixture(scope='function')
+def login_user(driver):
+    base_page = BasePage(driver)
```
Нужно исправить: объект BasePage никогда не создается, это не какая-то конкретная страница а интерфейс

AlexKitIt commented on conftest.py:
```diff
@@ -0,0 +1,56 @@
+def login_user(driver):
+    base_page = BasePage(driver)
+    base_page.open_page(urls.URL.STELLAR_BURGERS)
+    personal_account_click = base_page.wait_and_find_element(SBurgersLocators.PERSONAL_ACCOUNT_BUTTON)
```
Нужно исправить: здесь и далее: локаторов в тестах и фиксутрах быть не должно, если это не вариант параметризации, они живут в методах пейджи, тест должен быть максимально изолирован от логики выполнения

AlexKitIt commented on conftest.py:
```diff
@@ -0,0 +1,56 @@
+    base_page = BasePage(driver)
+    base_page.open_page(urls.URL.STELLAR_BURGERS)
+    personal_account_click = base_page.wait_and_find_element(SBurgersLocators.PERSONAL_ACCOUNT_BUTTON)
+    driver.execute_script("arguments[0].click();", personal_account_click)
```
Нужно исправить: здесь и далее: прямого обращения к driver из тестов и фикстур быть не должно , только через методы пейджей

AlexKitIt commented on pages/main_page.py:
```diff
@@ -0,0 +1,66 @@
+from seletools.actions import drag_and_drop
+
+class MainFunctional(BasePage):
+    @allure.step('клик по кнопке Личный кабинет')
+    def click_to_personal_account(self):
+        element = self.wait_and_find_element(locators.SBurgersLocators.PERSONAL_ACCOUNT_BUTTON)
+        self.driver.execute_script("arguments[0].click();", element)
```
Нужно исправить: здесь и далее по всему проекту: в пейдже описываются только методы работы с локаторами на конкретной странице, методы копипастятся из пейджи в пейджу, так быть не должно

LikeARivers committed:
удалены лишние комментарии
удалены дубли локаторов
пэйджи не используют дубли из других пэйджей
в фикстуре не создается BasePage
локаторы удалены из фикстур и тестов
прямое обращение к driver из тестов и фикстур убрано

AlexKitIt commented on pages/list_of_orgers_page.py:
```diff
@@ -0,0 +1,73 @@
+    @allure.step('Клик по заказу в ленте заказов')
+    def click_to_order(self):
+        element = self.wait_and_find_element(locators.SBurgersLocators.ORDER_ON_LIST)
+        self.driver.execute_script("arguments[0].click();", element)
```
Повторный коммент: Нужно исправить: здесь и далее по проекту: все прямые обращения к driver  необходимо вынести в base и оттуда за счет наследования переиспользовать в объектах страниц

LikeARivers committed:
удалено обращение к driver из pages
